src/util/postgres_client.js

```diff
@@ -149,13 +153,46 @@ function handle_ops_encoding(schema, val) {
 
 let query_counter = 0;
 
+async function log_query(pg_client, q, tag, millitook, pool_count, should_explain) {
+    const explain_q = { text: 'explain ' + q.text, values: q.values };
+    let explain_rows = ['explain:\tno explain'];
+    try {
+        if (should_explain && process.env.EXPLAIN_QUERIES === 'true') {
+            let explain_res = await pg_client.query(explain_q);
+            explain_rows = explain_res.rows.map(row => 'explain:\t' + row['QUERY PLAN']);
+        }
+    } catch (err) {
+        console.error('got error on explain', explain_q, err);
+        explain_rows = ['explain:\texplain failed'];
+    }
+
+    let stack = (new Error()).stack.split('\n').slice(3).join(',');
+    const msgs = [];
+    msgs.push(`stack: ${stack}`);
+    msgs.push(`tag: ${tag}`);
+    msgs.push(`query: ${q.text}`);
+    msgs.push(`took: ${millitook}`);
+    msgs.push(`pool: ${pool_count}`);
+    msgs.push(`explain:`);
+    let log_message = msgs.concat(explain_rows).join('\nQUERY_LOG:\t');
+    console.log(`\nQUERY_LOG:\t${log_message}`);
+}
+
+
 async function _do_query(pg_client, q, transaction_counter) {
     query_counter += 1;
     const tag = `T${_.padStart(transaction_counter, 8, '0')}|Q${_.padStart(query_counter.toString(), 8, '0')}`;
     try {
-        dbg.log1(`postgres_client: ${tag}: ${q.text}`, util.inspect(q.values, { depth: 6 }));
+        // dbg.log0(`postgres_client: ${tag}: ${q.text}`, util.inspect(q.values, { depth: 6 }));
+        const millistart = time_utils.millistamp();
+        const pool_count = `total=${pg_client.totalCount} waiting=${pg_client.waitingCount} idle=${pg_client.idleCount}`;
         const res = await pg_client.query(q);
-        dbg.log1(`postgres_client: ${tag}: got result:`, util.inspect(res, { depth: 6 }));
+        const millitook = time_utils.millitook(millistart);
+        try {
+            await log_query(pg_client, q, tag, millitook, pool_count, transaction_counter === 0);
```

Comment:
Can you add a comment to explain why `should_explain = transaction_counter === 0` ?
It's not clear to me.

Reply:
sure. 
In general the log_query code is a quick and dirty solution. I plan to refactor it if and before merging. the if is there since failures in the explain for transaction caused the transaction to no longer be valid, so I decided for now to just avoid it

Reply:
got it. I think it's worth keeping instrumentation inside the code behind env var, so we can use for next performance improves. or you have a better way to collect those from `pg_stat_statements`?
perhaps keep such scripts in git too?